tiledb/sm/query/readers/result_tile.cc

```diff
@@ -830,9 +830,8 @@ void ResultTile::compute_results_count_sparse_string_range(
     // dim.
     uint64_t count = 0;
     for (uint64_t j = start_range_idx; j < end_range_idx; ++j) {
-      auto& range = cached_ranges[j];
-      count += str_coord_intersects(
-          c_offset, c_size, buff_str, range.first, range.second);
+      const auto& range = cached_ranges[j];
+      count += str >= range.first && str <= range.second;
```

Comment:
Does pulling this out help or would the `const` annotation be enough? (keeping that computation in one function is appealing for future legibility/refactoring)

Reply:
The main issue was the str_coord_inrersects function was creating new string view objects even even thought we are looking on the ranges here not the cell so no need for new views. The const on cached_ranges was also a recommendation but I don't think it helps. The for loops can't be vectorized because of the two conditionals.